src/com/tictacgo/data/Board.java

```diff
@@ -154,99 +153,115 @@ public boolean isCatsGame() {
+      mergeWinners(getWinners(row), winners);
+    }
 
-        // Top left to bottom right diagonal
-        sum = 0;
-        int row = 0;
-        int col = 0;
-        while (row < SIDE_LENGTH && col < SIDE_LENGTH) {
-            ArrayList<Piece> space = pieces.get(row).get(col);
-            // Spaces with multiple pieces block winning.
-            if (space.size() == 1) {
-                sum += playerValues.get(space.get(0).getPlayer());
-            }
-            row++;
-            col++;
-        }
-        if (sum == winningSumX) {
-            winners.put(Player.X, winners.get(Player.X) + 1);
-        } else if (sum == winningSumY) {
-            winners.put(Player.O, winners.get(Player.O) + 1);
-        }
+    // Columns
+    for (int col = 0; col < SIDE_LENGTH; col++) {
+      ArrayList<Space> columnSpaces = new ArrayList<>(SIDE_LENGTH);
 
-        // Top right to bottom left diagonal
-        sum = 0;
-        row = SIDE_LENGTH - 1;
-        col = SIDE_LENGTH - 1;
-        while (row >= 0 && col >= 0) {
-            ArrayList<Piece> space = pieces.get(row).get(col);
-            // Spaces with multiple pieces block winning.
-            if (space.size() == 1) {
-                sum += playerValues.get(space.get(0).getPlayer());
-            }
-            row--;
-            col--;
-        }
-        if (sum == winningSumX) {
-            winners.put(Player.X, winners.get(Player.X) + 1);
-        } else if (sum == winningSumY) {
-            winners.put(Player.O, winners.get(Player.O) + 1);
-        }
+      for (ArrayList<Space> row : spaces) {
+        columnSpaces.add(row.get(col));
+      }
+
+      mergeWinners(getWinners(columnSpaces), winners);
+    }
+
+    // Top left to bottom right diagonal
+    int row = 0;
+    int col = 0;
+    ArrayList<Space> diagSpaces = new ArrayList<>(SIDE_LENGTH);
+    while (row < SIDE_LENGTH && col < SIDE_LENGTH) {
+        diagSpaces.add(spaces.get(row).get(col));
+        row++;
+        col++;
+    }
+
+    mergeWinners(getWinners(diagSpaces), winners);
+
+    // Top right to bottom left diagonal
+    diagSpaces.clear();
+    row = 0;
+    col = SIDE_LENGTH - 1;
+    while (row < SIDE_LENGTH && col >= 0) {
+        diagSpaces.add(spaces.get(row).get(col));
+        row++;
+        col--;
+    }
+
+    mergeWinners(getWinners(diagSpaces), winners);
 
 		return winners;
 	}
 
+  /**
+   * Figures out if there is a winner in the given Spaces. A winner must be in every single Space
+   * to win.
+   *
+   * @param toCheck A List of the Spaces to check for a winner. This should be an entire row,
+   *                column, or diagonal of Spaces.
+   * @return A Set of the Players who won in this List of Spaces. We don't need to map to a count
+   * here because a Player may only win once per row, column, or diagonal.
+   */
+  private Set<Player> getWinners(List<Space> toCheck) {
+    //TODO: This allows spaces with multiple Pieces in them to still contribute to a win.
```

Comment:
No, any pieces. The logic was simpler to code that way, but I left the TODO in case we want to revert that back.